public/app/features/transformers/calculateHeatmap/heatmap.ts

```diff
@@ -329,7 +327,7 @@ export function calculateHeatmapFromData(
     xMode: xBucketsCfg.mode,
     xSize:
       xBucketsCfg.mode === HeatmapCalculationMode.Size
-        ? durationToMilliseconds(parseDuration(xBucketsCfg.value ?? ''))
+        ? convertDurationToMilliseconds(xBucketsCfg.value ?? '')
```

Comment:
My options were either change parseDuration to support milliseconds or create a separate function to handle this. Since parseDuration's map is what's available in date-fns ([src](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/datetime/durationutil.ts#L3)), I decided to create my own function just for heatmap. If you disagree, we can change the core duration functions to support milliseconds as well, by making fractional seconds.